src/Neo/Network/P2P/Payloads/Signer.cs

```diff
@@ -190,16 +189,30 @@ void IInteroperable.FromStackItem(VM.Types.StackItem stackItem)
             throw new NotSupportedException();
         }
 
-        VM.Types.StackItem IInteroperable.ToStackItem(ReferenceCounter referenceCounter)
+        VM.Types.StackItem IInteroperable.ToStackItem(ApplicationEngine applicationEngine)
         {
-            return new VM.Types.Array(referenceCounter, new VM.Types.StackItem[]
+            if (!applicationEngine.IsHardforkEnabled(Hardfork.HF_Cockatrice))
```

Comment:
Do we have any change in behavior for on-chain transactions? I doubt you can make a transaction that would trigger this FAULT.

Reply:
Agree, it seems that this code may safely be handled without hardfork. And in general, I'd avoid making `ToStackItem` depend on `ApplicationEngine` as far as possible.